dig_test.go

```diff
@@ -344,6 +344,29 @@ func TestEndToEndSuccess(t *testing.T) {
 		}))
 	})
 
+	t.Run("ignore unexported fields", func(t *testing.T) {
+		type type1 struct{}
+		type type2 struct{}
+		type type3 struct{}
+		constructor := func() (*type1, *type2) {
+			return &type1{}, &type2{}
+		}
+
+		c := New()
+		type param struct {
+			In `ignore-unexported:"true"`
+
+			T1 *type1 // regular 'ol type
+			T2 *type2 `optional:"true"` // optional type present in the graph
+			t3 *type3
```

Comment:
can we have that or a different constructor produce `type3` so we can verify that the unexported field t3 is not set?